src/sidebar/components/Annotation/Annotation.js

```diff
@@ -18,6 +18,23 @@ import AnnotationReplyToggle from './AnnotationReplyToggle';
  * @typedef {import('../../../types/api').Group} Group
  */
 
+function SavingMessage() {
+  return (
+    <div
+      className={classnames(
+        'flex grow justify-end items-center gap-x-1',
+        // Make sure height matches that of action-bar icons so that there
+        // isn't a height change when transitioning in and out of saving state
+        'h-8 touch:h-touch-minimum'
+      )}
+      data-testid="saving-message"
+    >
+      <Spinner classes="text-xl" size="small" />
```

Comment:
I think I prefer the look without the spinner, at least when the save happens quickly (say < 500ms or so). Ideally we'd like the save to appear instant/near-instant and the spinner calls attention to the intermediate state a bit more. A couple of ideas related to this:

- Show the action bar but with disabled actions after the save, then show a saving indicator if it takes longer than say 500ms to save.
- Show only the "Saving..." text initially, and then show the spinner to give more visible activity indication if the save takes longer than say, 500ms.

Reply:
I appreciate the feedback! I'll give it a quick think.

> Show only the "Saving..." text initially, and then show the spinner to give more visible activity indication if the save takes longer than say, 500ms.

We could maybe approximate this with little fuss by `transition`-ing opacity on the `Spinner`.